app/anti_virus_check.py

```diff
@@ -0,0 +1,139 @@
+import collections
+import json
+import os
+import time
+
+import requests
+from sdc.rabbit.exceptions import QuarantinableError, RetryableError, BadMessageError
+
+from app import create_and_wrap_logger
+from app import settings
+
+logger = create_and_wrap_logger(__name__)
+
+AVResult = collections.namedtuple('AVResult', 'safe ready scan_results')
+
+
+class AntiVirusCheck:
+    def __init__(self, tx_id):
+        self.bound_logger = logger.bind(tx_id=tx_id)
+
+    def send_for_av_scan(self, payload):
+        self.bound_logger.debug("A/V scanned enabled {}".format(settings.ANTI_VIRUS_ENABLED))
+        self.bound_logger.info("Sending for AV check")
+        data_id = self._send_for_anti_virus_check(payload.file_name, payload.decoded_contents)
+        ready = False
+        attempts = 0
+        while not ready:
+            results = self._get_anti_virus_result(data_id)
+            ready = results.ready
+            if not ready:
+                attempts += 1
+                if attempts >= settings.ANTI_VIRUS_WAIT_TIME:
+                    self.bound_logger.error(
+                        "Unable to get results of Anti-virus scan for case id {} and file {} ".format(payload.case_id,
+                                                                                                      payload.file_name))
+                    raise RetryableError()
+                time.sleep(settings.ANTI_VIRUS_WAIT_TIME)
+            elif not results.safe:
+                error = "Unsafe file detected for case id {} with filename {}".format(payload.case_id,
+                                                                                      payload.file_name)
+                self._write_scan_report(results, payload.file_name)
+                self.bound_logger.error(error)
+                raise QuarantinableError()
+            else:
+                self.bound_logger.debug(
+                    "File {} for case id {} has been virus checked and confirmed safe".format(payload.case_id,
+                                                                                              payload.file_name))
+                return True
+
+    def _add_api_key(self, headers):
+        if settings.ANTI_VIRUS_API_KEY:
+            self.bound_logger.debug("Setting A/V API key")
+            headers['apikey'] = settings.ANTI_VIRUS_API_KEY
```

Comment:
Can we check the existence of the API key on startup if AV is enabled? We could then remove this check (and method) and set the API key in the headers dict in `_send_for_anti_virus_check`.